src/backend/optimizer/path/allpaths.c

```diff
@@ -3006,22 +3006,41 @@ recurse_push_qual(Node *setOp, Query *topquery,
 static bool
 is_query_contain_limit_groupby(Query *parse)
```

Comment:
May be we should rename this function?

Reply:
I would agree, now the name is a bit outdated